### PR TITLE
Add Anomali ThreatStream support to threatintel module

### PR DESCRIPTION
**Pull request by adriansr**
## What does this PR do?

This adds a new dataset, `anomalithreatstream` to the `threatintel` module.

## Why is it important?

Allows to ingest indicators from Anomali ThreatStream Integrator via a custom SDK output.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Screenshots

Docs:

image

### CODE REVIEW

**Review by P1llus — APPROVED** [review]
Looking great and the TI Mapping seems to be on point as well! Asked the CTI team to also confirm, but if there is no feedback from them then this is all good to go 👍 

Awesome work!

File: x-pack/filebeat/module/threatintel/anomali_threatstream/config/config.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  - add_fields:
+      target: ''
+      fields:
+        ecs.version: 1.10.0
+  - convert:
+      fields:
+        - from: json.id
+          to: '@metadata._id'
+          type: string
+      mode: copy
```
Comment (adriansr) [inline x-pack/filebeat/module/threatintel/anomali_threatstream/config/config.yml:50]: This is a bad idea as this ID is an incremental small number and can collide with other sources in the same index, convert to fingerprint using the ID and a unique field like `event.dataset`.

**Review by adriansr — COMMENTED** [review]
Something to fix

**Review by andrewkroh — APPROVED** [review]
LGTM. Nice docs.

Should this have its own dashboard?

**Comment by adriansr** [comment]
Renamed from `anomali_threatstream` to `anomalithreatstream` to prevent some issues with the new Threat Intel Dashboard.

**Pull request merged**